Code/Framework/AzCore/AzCore/azcore_files.cmake

```diff
@@ -34,6 +34,7 @@ set(FILES
     Asset/AssetTypeInfoBus.h
     Asset/AssetInternal/WeakAsset.h
     Casting/lossy_cast.h
+    Casting/numeric_cast_internal.h
     Casting/numeric_cast.h
```

Comment:
Sorting wise, wouldn't `numeric_cast_internal.h` come after `numeric_cast.h`?

Reply:
I was following VSCode which seems to not take into account the extension, will flip this